### PR TITLE
reduce postprocessors to `Function`

### PR DESCRIPTION
**Pull request by Shoeboxam**
This PR removes `AgnosticMetric` and changes the representation of postprocessors from `Transformation<DI, DO, AgnosticMetric, AgnosticMetric>` to `Function<TI, TO>`.

### CODE REVIEW

File: rust/src/core/ffi.rs [hunk 1]
```diff
@@ -550,6 +588,43 @@ pub extern "C" fn opendp_core__measurement_output_distance_type(this: *mut AnyMe
...
+
+#[bootstrap(
+    name = "_function_free",
+    arguments(this(do_not_convert = true)),
+    returns(c_type = "FfiResult<void *>")
+)]
+/// Internal function. Free the memory associated with `this`.
+#[no_mangle]
+pub extern "C" fn opendp_core___function_free(this: *mut AnyFunction) -> FfiResult<*mut ()> {
+    util::into_owned(this).map(|_| ()).into()
```
Comment (andrewvyrros) [inline rust/src/core/ffi.rs:625]: Always a treat to add another instance of the toilet closure!

**Review by andrewvyrros — COMMENTED** [review]
Another nice improvement. Using transformations for post processing as seemed right at the time, but this feels cleaner.

File: python/src/opendp/transformations.py [hunk 2]
```diff
@@ -630,21 +630,18 @@ def make_cast_inherent(
 
 def make_cdf(
     TA: RuntimeTypeDescriptor = "float"
-) -> Transformation:
+):
```
[the remark this answers is not in the document]
Comment (Shoeboxam) [reply python/src/opendp/transformations.py:633]: Fixed. It returns Function now, not Transformation.

File: python/src/opendp/mod.py [hunk 3]
```diff
@@ -86,10 +86,13 @@ def check(self, d_in, d_out, *, debug=False) -> bool:
                 return False
             raise
 
-    def __rshift__(self, other: "Transformation"):
-        if isinstance(other, Transformation):
-            from opendp.combinators import make_chain_tm
-            return make_chain_tm(other, self)
+    def __rshift__(self, other: "Function"):
+        if isinstance(other, (Transformation, Measurement)):
+            other = other.function
```
[the remark this answers is not in the document]
Comment (Shoeboxam) [reply python/src/opendp/mod.py:91]: Removed it. I just got carried away with the sugar.

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #648** <a href="https://app.graphite.dev/github/pr/opendp/opendp/648" target="_blank">Graphite</a>  👈
    * **PR #618** <a href="https://app.graphite.dev/github/pr/opendp/opendp/618" target="_blank">Graphite</a> 
      * **PR #456** <a href="https://app.graphite.dev/github/pr/opendp/opendp/456" target="_blank">Graphite</a> 
        * **PR #641** <a href="https://app.graphite.dev/github/pr/opendp/opendp/641" target="_blank">Graphite</a> 
        * **PR #653** <a href="https://app.graphite.dev/github/pr/opendp/opendp/653" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/648?utm_source=stack-comment).

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #648** <a href="https://app.graphite.dev/github/pr/opendp/opendp/648" target="_blank">Graphite</a>  👈
    * **PR #618** <a href="https://app.graphite.dev/github/pr/opendp/opendp/618" target="_blank">Graphite</a> 
      * **PR #456** <a href="https://app.graphite.dev/github/pr/opendp/opendp/456" target="_blank">Graphite</a> 
        * **PR #641** <a href="https://app.graphite.dev/github/pr/opendp/opendp/641" target="_blank">Graphite</a> 
        * **PR #653** <a href="https://app.graphite.dev/github/pr/opendp/opendp/653" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/648?utm_source=stack-comment).

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**